runtime/tests/interpreter/string_test.go

```diff
@@ -42,13 +42,13 @@ func TestInterpretRecursiveValueString(t *testing.T) {
 	require.NoError(t, err)
 
 	require.Equal(t,
-		`{"mapRef": {"mapRef": ...}}`,
+		`{"mapRef": {"mapRef": {"mapRef": ...}}}`,
 		mapValue.String(interpreter.StringResults{}),
 	)
 
 	require.IsType(t, &interpreter.DictionaryValue{}, mapValue)
 	require.Equal(t,
-		`{"mapRef": ...}`,
+		`{"mapRef": {"mapRef": ...}}`,
```

Comment:
Just curious as to what causes the additional iteration when converting to string? :thinking:

Reply:
It was late last evening, but I also was curious why this changed, so I quickly looked into it: 
This is caused by the change of the `Copy` function of the `EphemeralReferenceValue`, which used to return the same pointer, but now allocated a new one. I changed it to be more defensive, though technically it's not needed, as there are no mutating functions